_docs/setup/kubernetes/mesh-expansion.md

```diff
@@ -0,0 +1,208 @@
+   ```
+   ```
+   ISTIO_SERVICE_CIDR=10.23.240.0/20
+   ```
+* Generate DNS config file to be used in the VMs. This will allow apps on the VM to resolve
+cluster service names, which will be intercepted by the sidecar and forwarded.
+
+```bash
+install/tools/setupMeshEx generateConfigs MY_CLUSTER_NAME
+```
+
+Example generated files:
+
+   ```bash
+cat /etc/dnsmasq.d/kubedns
+```
+```
+   server=/svc.cluster.local/10.128.0.6
+   address=/istio-mixer/10.128.0.7
+   address=/mixer-server/10.128.0.7
+   address=/istio-pilot/10.128.0.5
+   address=/istio-ca/10.128.0.8
+  ```
+
+### Setting up the machines
+
+* Copy the config files and istio debian files to each machine joining the cluster, as
+`/etc/dnsmasq.d/kubedns` and `/var/lib/istio/envoy/cluster.env`.
+
+* Configure and verify DNS settings - this may require installing dnsmasq, adding it to
+`/etc/resolv.conf` directly or via DHCP scripts. To verify, check that the VM can resolve
+names and connect to pilot, for example:
+
+On the VM/external host:
+```bash
+dig istio-pilot.istio-system
+```
+```
+    # This should be the same address shown as "EXTERNAL-IP" in 'kubectl get svc -n istio-system istio-pilot-ilb'
+    ...
+    istio-pilot.istio-system. 0	IN	A	10.128.0.5
+    ...
+```
+```bash
+    # Check that we can resolve cluster IPs. The actual IN A will depend on cluster config.
+    dig istio-pilot.istio-system.svc.cluster.local.
+```
+```
+    ...
+    istio-pilot.istio-system.svc.cluster.local. 30 IN A 10.23.251.121
+```
+```bash
+dig istio-ingress.istio-system.svc.cluster.local.
+```
+```
+    ...
+    istio-ingress.istio-system.svc.cluster.local. 30 IN A 10.23.245.11
+```
+
+* Verify connectivity. We'll check if the VM can connect to the pilot, and it can connect to
+an endpoint.
+
+    ```bash
+    curl -v 'http://istio-pilot.istio-system:8080/v1/registration/istio-pilot.istio-system.svc.cluster.local|http-discovery'
+    ```
+    ```
+    ...
+    "ip_address": "10.20.1.18",
+    ...
+    ```
+    ```bash
+    # On the VM: use the address above - it will connect directly the the pod running istio-pilot.
+    curl -v 'http://10.20.1.18:8080/v1/registration/istio-pilot.istio-system.svc.cluster.local|http-discovery'
+    ```
+
+* Extract the initial Istio authentication secrets and copy them to the machine. The default
+installation of Istio includes IstioCA and will generate Istio secrets even if automatic 'mTLS'
+setting is disabled. It is recommended that you perform this step to make it easy to
+enable mTLS in future and upgrade to future version that will have mTLS enabled by default.
+
+```bash
+  # ACCOUNT defaults to 'istio.default', or SERVICE_ACCOUNT environment variable
+  # NAMESPACE defaults to current namespace, or SERVICE_NAMESPACE environment variable
+
+  install/tools/setupMeshEx machineCerts ACCOUNT NAMESPACE
+```
+
+The generated files (key.pem, root-cert.pem, cert-chain.pem) must be copied to /etc/certs on each machine, readable by istio-proxy.
```

Comment:
put file names and the file path into code style

Reply:
done